Libraries/libcar/Sources/Rendition.cpp

```diff
@@ -250,6 +248,10 @@ Load(
     rendition.layout() = layout;
     rendition.resizeMode() = ResizeModeFromLayout(layout);
 
+    if (rendition.slices().size() == 0) {
+        rendition.slices().push_back({0, 0, value->width, value->height});
```

Comment:
What about doing this in the constructor? That way the `write()` method can write out exactly what's in the object, even if it's technically invalid, and it's still easy to get it right for clients.